### ISSUE TITLE
java.lang.ExceptionInInitializerError

### ISSUE THREAD

**Issue by fahadDeveloper** [koral--/android-gif-drawable#237]
I am using this library. it was working fine I just add this compile 'pl.droidsonroids.gif:android-gif-drawable:1.1.+' in my build.gradle and all was going fine. but when I tried to add google play services 
compile 'com.google.android.gms:play-services-auth:8.3.0' it was giving me  java.lang.UnsatisfiedLinkError I read your issue on link https://github.com/koral--/android-gif-drawable/issues/51#issuecomment-155120328 then I added 'pl.droidsonroids.gif:android-gif-drawable:1.1.11' in build.gradle and mavenCentral() in project level build.gradle and I've used LibraryLoader.initialize(this.getApplicationContext()); in different places in my app . Then it started to give me this error stack trace:
java.lang.ExceptionInInitializerError
     at pl.droidsonroids.gif.GifDrawable.<init>(GifDrawable.java:161)
     at pl.droidsonroids.gif.GifDrawable.<init>(GifDrawable.java:96)
     at pl.droidsonroids.gif.GifViewUtils.setResource(GifViewUtils.java:54)
     at pl.droidsonroids.gif.GifImageView.setImageResource(GifImageView.java:101)
     at myapp.mypackage.tour.SlideFragment$1.run(SlideFragment.java:105)
     at android.os.Handler.handleCallback(Handler.java:739)
     at android.os.Handler.dispatchMessage(Handler.java:95)
     at android.os.Looper.loop(Looper.java:148)
     at android.app.ActivityThread.main(ActivityThread.java:5417)
     at java.lang.reflect.Method.invoke(Native Method)
     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
  Caused by: java.lang.IllegalStateException: Library libpl_droidsonroids_gif.so for supported ABIs not found in APK file
     at pl.droidsonroids.gif.ReLinker.getLibraryEntry(ReLinker.java:143)
     at pl.droidsonroids.gif.ReLinker.unpackLibrary(ReLinker.java:97)
     at pl.droidsonroids.gif.ReLinker.loadLibrary(ReLinker.java:57)
     at pl.droidsonroids.gif.LibraryLoader.loadLibrary(LibraryLoader.java:49)
     at pl.droidsonroids.gif.GifInfoHandle.<clinit>(GifInfoHandle.java:42)
     at pl.droidsonroids.gif.GifDrawable.<init>(GifDrawable.java:161) 
     at pl.droidsonroids.gif.GifDrawable.<init>(GifDrawable.java:96) 
     at pl.droidsonroids.gif.GifViewUtils.setResource(GifViewUtils.java:54) 
     at pl.droidsonroids.gif.GifImageView.setImageResource(GifImageView.java:101) 
     at myapp.mypackage.tour.SlideFragment$1.run(SlideFragment.java:105) 
     at android.os.Handler.handleCallback(Handler.java:739) 
     at android.os.Handler.dispatchMessage(Handler.java:95) 
     at android.os.Looper.loop(Looper.java:148) 
     at android.app.ActivityThread.main(ActivityThread.java:5417) 
     at java.lang.reflect.Method.invoke(Native Method) 
     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 

please help

**Comment by koral--** [comment]
` java.lang.IllegalStateException: Library libpl_droidsonroids_gif.so for supported ABIs not found in APK file` means that there is no directory for expected ABI inside APK file.
One of the reasons is that you have defined ABI filter or ABI split in `build.gradle` and installed APK does not match your device.
If you have not defined split nor filter please post what is your device and what is the value of `android.os.Build.CPU_ABI` and if it Lollipop or newer `android.os.Build.SUPPORTED_ABIS`.

**Comment by fahadDeveloper** [comment]
I've search my build.gradle (both) couldn't find anything like ABI filter or ABI split, I've printed like this, Log.d(TAG, "CPU_ABI: " +Build.CPU_ABI + ", CPU_ABI2: " + Build.CPU_ABI2 + ", Device: " + Build.DEVICE); it gives me this output:
CPU_ABI: armeabi-v7a, CPU_ABI2: armeabi, Device: hammerhead
and my device is LGE Nexus 5 Android 6.0.1 (API 23)

**Comment by koral--** [comment]
OK, `CPU_ABI` is correct for Nexus 5. Could you check if file `lib/armeabi-v7a/libpl_droidsonroids_gif.so` is located inside your APK? You can do that using commadline eg. `unzip -l <path to apk> |grep armeabi-v7a` (file should be listed) or using some file manager (open apk like zip file).

**Comment by fahadDeveloper** [comment]
when I run my app with google play api I couldn't find even the lib folder in my apk. but when I build and run without that apk all is well.

**Comment by koral--** [comment]
So it looks like that something is preventing native libs directory from being bundled into APK. Are using latest build tools and android gradle plugins?
I've tried applying steps from https://developers.google.com/cloud-messaging/android/client#add-config to the clean sample project https://github.com/koral--/android-gif-drawable-sample and native libraries are packaged correctly.

**Comment by fahadDeveloper** [comment]
thank you koral. I changed my buildToolsVersion to "23.0.2", then I include latest google plus api compile 'com.google.android.gms:play-services-auth:8.4.0' in app level build.gradle and in project level build.gradle:
  classpath 'com.android.tools.build:gradle:2.0.0-alpha3'
   classpath 'com.google.gms:google-services:2.0.0-alpha3'
then I also have change distributionUrl in gradle-wrapper.properties file to https\://services.gradle.org/distributions/gradle-2.8-all.zip
then build and run. now everything works I also got signed in from google. thank you sir really appreciated
